src/react/utils.js

```diff
@@ -414,3 +436,30 @@ export function getProperty(realm: Realm, object: ObjectValue, property: string
   invariant(value instanceof Value, `react/utils/getProperty should not be called on internal properties`);
   return value;
 }
+
+// if a render prop function (a nested additional function) makes
+// no accesses to bindings in the parent additional function scope
+// we can determine if the function is self contained
+export function isRenderPropFunctionSelfContained(
+  realm: Realm,
+  parentFunc: Value,
+  renderProp: FunctionValue,
+  logger: any // otherwise Flow cycles increases
+) {
+  let { unboundReads, unboundWrites } = getLeakedFunctionInfo(renderProp);
```

Comment:
I wonder if this could be expressed similarly to `reportWriteConflicts` in `functions.js`. That function instruments all property reads/writes to report when two additional functions modify the same properties. I forget if it also does something for bindings as well.